OmgvaPOS/src/ItemManagement/Services/ItemService.cs

```diff
@@ -78,5 +84,25 @@ public void DeleteItem(long id) {
             //TODO: Delete from open orders?
             //TODO: Think about discounts
         }
+
+        public List<TaxDto> GetItemTaxes(long id) {
+            var taxItems = _taxItemRepository.GetAllTaxItemQueriable().Where(c => c.ItemId == id);
+            return _taxRepository.GetAllTaxes()
+                .Where(t => taxItems.Select(c => c.TaxId).Contains(t.Id))
+                .Select(t => TaxMapper.ToDTO(t)).ToList();
+        }
+        public ItemDTO ChangeItemTaxes(ChangeItemTaxesRequest changeItemTaxesRequest, long itemId) {
+            var newItem = UpdateItem(_itemRepository.GetItem(itemId)); //TODO: Potential error here
```

Comment:
What does this part do? Based on just this line, we get an item from repository and then call UpdateItem with the same Item model without any extra arguments or modifications, please rewrite

Reply:
What this essentially does, it just archives a copy of the item and creates a new one, as well as adjusting all the relevant foreign keys and creating copies of items' variations.